src/cpp/core/FileInfo.cpp

```diff
@@ -43,6 +43,8 @@ FileInfo::FileInfo(const std::string& absolutePath,
       lastWriteTime_(0),
       isSymlink_(isSymlink)
 {
+   // some file paths might be constructed with trailing nul bytes; remove those here
```

Comment:
Out of curiosity, what's resulting in trailing `\0` s?

Reply:
I had this change in my mind (although this should be fixed now):

https://github.com/rstudio/rstudio/commit/1e3bed6b8034584a0d462e5c550af9fbf0440d18

But the fact that we were calling `.c_str()` before made me think we need to be cautious about any other potential strings coming in with trailing null bytes.

Reply:
I'm still not clear on this; a `std::string` initialized from a C string (e.g. `char *`) always stores the trailing \0. So I think absolutePath_ = absolutePath_.c_str() is a noop?

Reply:
Right :-) And we were constructing strings with a length in that particular spot, and I recall seeing that this was actually something like a "size of allocated space for file path" rather than "actual length of file path" in some caes.